src/collectors/mesos/mesos.py

```diff
@@ -4,70 +4,259 @@
 
 Collects metrics from a mesos instance. By default,
 the collector is set up to query the mesos-master via
-port 5050. Set the port to 5051 for mesos-slaves.
+port 5050. Set the port to 5051 for mesos-agent.
 
 #### Example Configuration
 
 ```
-    host = localhost
-    port = 5050
+host = localhost
+port = 5050
 ```
+
+#### Dependencies
+ * urlib2
 """
 
-import diamond.collector
-import json
+import copy
 import urllib2
-from urlparse import urlparse
+
+try:
+    import json
+except ImportError:
+    import simplejson as json
+
+import diamond.collector
+
+from diamond.collector import str_to_bool
 
 
 class MesosCollector(diamond.collector.Collector):
+    def __init__(self, config=None, handlers=[], name=None, configfile=None):
+        self.master = True
```

Comment:
probably don't need this since this is the default that is set under `get_default_config`